lark/exceptions.py

```diff
@@ -163,31 +192,35 @@ class UnexpectedToken(ParseError, UnexpectedInput):
 
     see: :ref:`ParserPuppet`.
     """
-    def __init__(self, token, expected, considered_rules=None, state=None, puppet=None, token_history=None):
+
+    def __init__(self, token, expected, considered_rules=None, state=None, puppet=None, all_terminals=None, token_history=None):
         # TODO considered_rules and expected can be figured out using state
         self.line = getattr(token, 'line', '?')
         self.column = getattr(token, 'column', '?')
         self.pos_in_stream = getattr(token, 'pos_in_stream', None)
         self.state = state
 
         self.token = token
-        self.expected = expected     # XXX deprecate? `accepts` is better
+        self.expected = expected  # XXX deprecate? `accepts` is better
         self.considered_rules = considered_rules
         self.puppet = puppet
+        self._all_terminals = all_terminals
         self.token_history = token_history
 
-        # TODO Only calculate `accepts()` when we need to display it to the user
```

Comment:
You removed this comment, but didn't change the computation.

`accepts()` is still being called even if it will never be displayed to the user.

Except now, repeated calls to `accepts()` will re-compute it every time instead of just once, so it's actually worse now. (that can be solved with memoization, but then we're back where we started)

Reply:
Better now?